### PR TITLE
Feature/deal with swtch stmt

### LINKED ISSUE #1: Deal with the edge case in switch-case syntax.
`xerrchk` can point out binary expressions comparing errors with ops such as `!=` and `==`. Such an expressions can lead to failure of error handling because wrapped errors should be unwrapped before comparison.
  Also, the switch-case syntax leads to the same failure of handling errors, and some cases described below should be checked too.

```
func switchErrCase1(err error) string {
	switch err { // want `do not compare error with switch-case syntax as it leads to the failure of handling specific errors.`
	case sentinelErr:
		return "true"
	default:
		return "false"
	}
}

func switchErrCase2(err error) string {
	switch TestCauseFunc(err) { // OK
	case sentinelErr:
		return "true"
	default:
		return "false"
	}
}

func TestCauseFunc(err error) error {
	return error
}
```

### PR DESCRIPTION
**Pull request by Matts966**
fix #1

### CODE REVIEW

File: passes/erreq/erreq.go [hunk 1]
```diff
@@ -50,7 +70,56 @@ func run(pass *analysis.Pass) (interface{}, error) {
 			return
 		}
 
-		pass.Reportf(node.Pos(), "do not compare error with \"==\" or \"!=\"")
```
Comment (Matts966) [inline passes/erreq/erreq.go]: Tests didn't pass in my local environment maybe because the format of the report.

File: passes/erreq/erreq.go [hunk 2]
```diff
@@ -11,27 +12,47 @@ import (
...
 }
 
-const Doc = "erreq finds binary expression which is comparing error"
-
 var errType = types.Universe.Lookup("error").Type().Underlying().(*types.Interface)
 
+var causeFuncs = []string{
+	"TestCauseFunc", // For test
+
+	"github.com/pkg/errors.Cause",
```
Comment (sachaos) [inline passes/erreq/erreq.go]: I want to focus on xerrors only, in this project.
So please remove this one.

File: passes/erreq/erreq.go [hunk 3]
```diff
@@ -11,27 +12,47 @@ import (
...
 		inspect.Analyzer,
 	},
 }
 
-const Doc = "erreq finds binary expression which is comparing error"
-
 var errType = types.Universe.Lookup("error").Type().Underlying().(*types.Interface)
 
+var causeFuncs = []string{
+	"TestCauseFunc", // For test
```
Comment (sachaos) [inline passes/erreq/erreq.go]: We can test with `xerrors.Unwrap`.
So please remove this one.

**Review by sachaos — CHANGES_REQUESTED** [review]

File: passes/erreq/erreq.go (hunk 2 above)
Comment (Matts966) [reply passes/erreq/erreq.go]: done

File: passes/erreq/erreq.go (hunk 3 above)
Comment (Matts966) [reply passes/erreq/erreq.go]: done

**Review by sachaos — APPROVED** [review]

**Pull request merged**